experiment/process-e2e-suite-log/main.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// input file path.
+var inputFilePathPtr *string
+
+// separator line to detect each section of the file.
+var separatorLine string = "----------------"
+
+// prefix used for the sub files generated.
+var prefixForSubFilePtr *string
+
+// tells if splitting step has to be processed or not.
+var skipSplittingPtr *bool
+
+// the quantity of workers to use
+var workersQuantityPtr *int
+
+// to access the endpoints map in an repeteable manner we need an ordered array.
+var keys []string
+
+// result file name.
+var resultFileNamePtr *string
+
+// separator character to use in result file.
+var separatorCharacterPtr *string
+
+// this replacer is to escape the text picked in the test description, in order to replace the ocurrences of
+// the character separator by space and not break the columns
+var replacer *strings.Replacer
+
+func createHeader() string {
+	var headerBuffer bytes.Buffer
+	headerBuffer.WriteString("filename")
+	headerBuffer.WriteString(*separatorCharacterPtr)
+	headerBuffer.WriteString("testDescription")
+
+	for _, key := range keys {
+		headerBuffer.WriteString(*separatorCharacterPtr)
+		headerBuffer.WriteString(key)
+	}
+	return headerBuffer.String()
+}
+
+func generateResultFile() {
+	AddRow(createHeader())
+
+	root, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	filepath.Walk(root, func(path string, f os.FileInfo, _ error) error {
+		if !f.IsDir() {
+			match, err := regexp.MatchString(*prefixForSubFilePtr, f.Name())
+			if err == nil && match {
+				AddFile(f.Name())
+			}
+		}
+		return nil
+	})
+
+	WaitUntilWorkGroupFinished()
+	WaitUntilResultWriterFinished()
+}
+
+// regex to pick individual lines from the complete text of the section.
+// This will be later used to pick specific lines to extract test description.
+var lineRegEx *regexp.Regexp = regexp.MustCompile(`(?m)^.*$`)
+
+// process individual file and generate a row.
+func processFile(fileName string) {
+	fileContent, err := ioutil.ReadFile(fileName)
+
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	var rowBuffer bytes.Buffer
```

Comment:
We could be less error-prone if we define a results struct and JSON serialization tags for it, so that we can use the Go standard library to output the data instead of maintaining a custom serializer.

Reply:
The order in which each column is printed is important and has to be consistent through all the execution (alongside with the header). Took a look at the json tags, could not find a straightforward way of guaranteeing order. Thanks for bringing this up. From this comment gave a second thought on it and set the process apart in 2 stages, using a rowData and rowMapper to split responsibilities aside. Let me know if it makes sense.